src/momento/cache_operation_types.py

```diff
@@ -163,6 +163,124 @@ def values_as_bytes(self) -> List[Optional[bytes]]:
         return r_values
 
 
+# Dictionary related respnses
+DictionaryKey = Union[str, bytes]
+DictionaryValue = Union[str, bytes]
+Dictionary = Dict[DictionaryKey, DictionaryValue]
+
+
+class CacheDictionaryGetResponse:
+    def __init__(self, value: Optional[DictionaryValue], result: CacheGetStatus):
+        self._value = value
+        self._result = result
+
+    def value(self) -> Optional[str]:
+        if self.status() == CacheGetStatus.MISS:
+            return None
+        return cast(bytes, self._value).decode("utf-8")
+
+    def value_as_bytes(self) -> Optional[bytes]:
+        if self.status() == CacheGetStatus.MISS:
+            return None
+        return cast(bytes, self._value)
+
+    def status(self) -> CacheGetStatus:
+        return self._result
+
+    def __str__(self) -> str:
+        return self.__repr__()
+
+    def __repr__(self) -> str:
+        value = self._value
+        try:
+            value = self.value()
+        except UnicodeDecodeError:
+            value = str(value)
+
+        return f"CacheDictionaryGetResponse(value={value}, result={self._result})"
+
+
+class CacheDictionaryValue:
+    def __init__(self, value: DictionaryValue) -> None:
+        self._value = value
+
+    def value(self) -> str:
+        return cast(bytes, self._value).decode("utf-8")
+
+    def value_as_bytes(self) -> bytes:
+        return cast(bytes, self._value)
+
+    def __eq__(self, other: object) -> bool:
+        return type(other) == type(self) and self._value == other._value  # type: ignore
+
+    def __str__(self) -> str:
+        return self.__repr__()
+
+    def __repr__(self) -> str:
+        value = self._value
+        try:
+            value = self.value()
+        except UnicodeDecodeError:
+            pass
```

Comment:
nit: in python you can just
```
try:
  value = self.value()
except UnicodeDecodeError:
  value = self._value
return f"... {value}"
```

here and elsewhere

Reply:
That's how I had it originally. The mypy (type checking) settings are overly strict and I had to write it like this to pass. Another thing to revisit.